test/IP3Token.EthlessReserve.test.js

```diff
@@ -59,10 +59,153 @@ describe('IP3Token - Ethless Reserve functions', function () {
+                nonce,
+                expirationBlock,
+                signature
+            );
+            await TestHelper.checkResult(input, IP3Token.address, user3, ethers, provider, 0);
+            expect(await IP3Token.balanceOf(owner.address)).to.equal(
+                ethers.BigNumber.from(originalBalance).sub(amountToReserve).sub(feeToPay)
+            );
+
+            const inputExecute = await IP3Token.connect(owner).populateTransaction['execute(address,uint256)'](
+                owner.address,
+                nonce
+            );
+            await TestHelper.checkResult(inputExecute, IP3Token.address, owner, ethers, provider, 0);
+            expect(await IP3Token.balanceOf(owner.address)).to.equal(
+                ethers.BigNumber.from(originalBalance).sub(amountToReserve)
+            );
+            expect(await IP3Token.balanceOf(user1.address)).to.equal(ethers.BigNumber.from(amountToReserve));
+        });
+        it('Test Ethless Reserve & Reclaim', async () => {
+            const originalBalance = await IP3Token.balanceOf(owner.address);
+
+            const nonce = Date.now();
+            const blockNumber = await provider.getBlockNumber();
+            const expirationBlock = blockNumber + 2000;
+
+            const signature = SignHelper.signReserve(
+                4,
+                network.config.chainId,
+                IP3Token.address,
+                owner.address,
+                owner.privateKey,
+                user1.address,
+                owner.address,
+                amountToReserve,
+                feeToPay,
+                nonce,
+                expirationBlock
+            );
+            const input = await IP3Token.connect(user3).populateTransaction[
+                'reserve(address,address,address,uint256,uint256,uint256,uint256,bytes)'
+            ](
+                owner.address,
+                user1.address,
+                owner.address,
+                amountToReserve,
+                feeToPay,
+                nonce,
+                expirationBlock,
+                signature
+            );
+            await TestHelper.checkResult(input, IP3Token.address, user3, ethers, provider, 0);
+            expect(await IP3Token.balanceOf(owner.address)).to.equal(
+                ethers.BigNumber.from(originalBalance).sub(amountToReserve).sub(feeToPay)
+            );
+
+            const inputReclaim = await IP3Token.connect(owner).populateTransaction['reclaim(address,uint256)'](
+                owner.address,
+                nonce
+            );
+            await TestHelper.checkResult(inputReclaim, IP3Token.address, owner, ethers, provider, 0);
+            expect(await IP3Token.balanceOf(owner.address)).to.equal(ethers.BigNumber.from(originalBalance));
+            expect(await IP3Token.balanceOf(user1.address)).to.equal(ethers.BigNumber.from(0));
+        });
+    });
+
+    describe('IP3Token - Test expecting failure Ethless Reserve', async function () {
```

Comment:
it is the failed case when the nonce was used only, try to add tests when different expiration, fee and amount + after expiration